### PR TITLE
feat: migrated the home page

### PR DESCRIPTION
**Pull request by khawarizmus**
This Pr should migrate the home page, Here are the main milestones:

- Migrate the Video Player
- Use Strapi to pull the featured campaigns
- Migrate the rest of the component to use composition API and script setup

### CODE REVIEW

File: components/global/GlobalFooter.vue [hunk 1]
```diff
@@ -4,65 +4,34 @@
...
     <div class="py-12 mx-12 sm:px-6 lg:py-16 lg:px-8">
       <div class="xl:grid xl:grid-cols-3 xl:gap-8">
-        <div class="space-y-8 xl:col-span-1">
-          <img
-            class="h-14"
-            src="~/assets/media/img/logo.svg"
-            alt="Company name"
-          />
+        <div class="space-y-8 mt-14 sm:mt-0 xl:col-span-1">
+          <img class="h-14" src="~/assets/media/img/logo.svg" alt="Company name" />
```
Comment (khawarizmus) [inline components/global/GlobalFooter.vue]: Nitpick: change `alt="Company name"` to `alt="An-nadaa logo"`

File: components/home/CampaignsSection.vue [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
-  },
-})
+// const props = defineProps({
+//   causes: {
+//     type: Array<Object>,
+//     required: true,
+//   },
+// })
+
+// const props = withDefaults(defineProps(), {
```
Comment (khawarizmus) [inline components/home/CampaignsSection.vue:20]: Not sure why prefer the flat syntax as it removed type inference and removed the default value of empty array.

File: components/home/FAQSection.vue [hunk 3]
```diff
@@ -35,20 +35,13 @@
...
-}
+const answers = ref<string[]>([])
 
-const props = defineProps({
-  faqs: {
-    type: Array<FAQ>,
-    required: true,
-  },
-})
+const props = defineProps(['faqs'])
```
Comment (khawarizmus) [inline components/home/FAQSection.vue:44]: Same. Not sure why prefer the flat syntax as it removed type inference and removed the default value of empty array.

File: components/home/TestimonialsSection.vue [hunk 4]
```diff
@@ -94,27 +93,19 @@
...
-}
+const plugins = [new Pagination({ type: 'bullet' })]
 
-const props = defineProps({
-  testimonials: {
-    type: Array<Testimonial>,
-    required: true,
-  },
-})
+const props = defineProps(['testimonials'])
```
Comment (khawarizmus) [inline components/home/TestimonialsSection.vue:107]: Same. Not sure why prefer the flat syntax as it removed type inference and removed the default value of empty array.

**Review by khawarizmus — COMMENTED** [review]
LGTM. a nitpick and few comments as a question.

File: components/home/TestimonialsSection.vue (hunk 4 above)
Comment (fazaamal) [reply components/home/TestimonialsSection.vue:107]: I was the one that changed it to the other syntax initially IIRC, and created my own types, which i thought were correct, but actually weren't. The props that are passed are actually nuxt content types, so they weren't actually compatible.

**Pull request merged**